developer/tutorials/articles/03-upgrading-code-to-liferay-7.2/01-intro.markdown

```diff
@@ -0,0 +1,233 @@
+---
+header-id: upgrading-code-to-product-ver
+---
+
+# Upgrading Code to @product-ver@
+
+[TOC levels=1-4]
+
+Upgrading to @product-ver@ involves migrating your installation and code (your
+custom apps) to the new version. You'll learn how to upgrade your code in this
+section.
+
+These tutorials assume you're using the
+[Code Upgrade Planner](/docs/reference/7-2/-/knowledge_base/r/code-upgrade-planner).
+To follow along with this section, install the planner and step through the
+upgrade instructions.
+
+For convenience, this tutorial section also references documentation and
+outlined steps to aid those opting to upgrade their code manually.
+
+Here are the code upgrade steps:
+
+1.  {.root} [Upgrade Your Development Environment](/docs/tutorials/7-2/-/knowledge_base/t/upgrading-your-development-environment) {.title}
+
+    Legacy project environments should be upgraded to the latest version of
+    Liferay Workspace to ensure you leverage all available features. {.summary}
+
+    1.  [Set Up Liferay Workspace](/docs/tutorials/7-2/-/knowledge_base/t/upgrading-your-development-environment#setting-up-liferay-workspace) {.title}
+
+        A Liferay Workspace is a generated environment that is built to hold
+        and manage your Liferay projects. Create/import a workspace to get
+        started. {.summary}
+
+        1.  [Create New Liferay Workspace](/docs/tutorials/7-2/-/knowledge_base/t/upgrading-your-development-environment#creating-new-liferay-workspace) {.title, commandId=create_new_liferay_workspace}
```

Comment:
hey @codyhoag there should be end with , (comma)

Reply:
@gamerson @jtydhr88 let me know how to format an attribute with multiple values. Should it be like this: `{.title commandId=create_new_liferay_workspace}`?